From 21a6b4dc8860f990ed7d9a957767f211b7bbda1f Mon Sep 17 00:00:00 2001
From: xhe <xw897002528@gmail.com>
Date: Fri, 7 Dec 2018 21:25:42 +0800
Subject: [PATCH 17/48] msgfmt: add missing '+' operator

---
 src/msgfmt.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/msgfmt.c b/src/msgfmt.c
index 4978f54..6fe78d0 100644
--- a/src/msgfmt.c
+++ b/src/msgfmt.c
@@ -88,7 +88,7 @@ int process_line_callback(po_message_t msg, void* user) {
 		m += msg->id_len + 1;
 
 		if (msg->plural_len)
-			m = msg->plural_len + 1;
+			m += msg->plural_len + 1;
 
 		if (msg->ctxt_len)
 			m += msg->ctxt_len + 1;
-- 
2.25.0

