From 29baed6eb3242e88465ac61b908ee4cac93d5cdc Mon Sep 17 00:00:00 2001
From: xhe <xw897002528@gmail.com>
Date: Fri, 7 Dec 2018 21:39:20 +0800
Subject: [PATCH 18/48] msgfmt: add missing messages count variable

---
 src/msgfmt.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/msgfmt.c b/src/msgfmt.c
index 6fe78d0..c464620 100644
--- a/src/msgfmt.c
+++ b/src/msgfmt.c
@@ -135,6 +135,8 @@ int process_line_callback(po_message_t msg, void* user) {
 				trans->len += m;
 				d->len[1] += m;
 			}
+
+			d->cnt++;
 		}
 
 		break;
-- 
2.25.0

