From b5136fecf3d73b072b622f2102af6d38cd2b5e07 Mon Sep 17 00:00:00 2001
From: xhe <xw897002528@gmail.com>
Date: Sun, 16 Dec 2018 15:13:08 +0800
Subject: [PATCH 22/48] poparser: avoid comparisons of different signs

---
 src/poparser.c | 4 ++--
 src/poparser.h | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/poparser.c b/src/poparser.c
index 7f87776..e71e386 100644
--- a/src/poparser.c
+++ b/src/poparser.c
@@ -162,7 +162,7 @@ enum po_error poparser_feed_line(struct po_parser *p, char* in, size_t in_len) {
 			line_len = len + 1;
 			x = p->buf;
 			len = p->bufsize;
-			if ( (len = iconv(p->cd, &line, &line_len, &x, &len)) == -1) 
+			if ( (len = iconv(p->cd, &line, &line_len, &x, &len)) == (size_t)-1) 
 				return -po_failed_iconv;
 
 			x = p->buf;
@@ -222,7 +222,7 @@ enum po_error poparser_feed_line(struct po_parser *p, char* in, size_t in_len) {
 			line_len = len + 1;
 			x = p->buf;
 			len = p->bufsize;
-			if ( (len = iconv(p->cd, &line, &line_len, &x, &len)) == -1) 
+			if ( (len = iconv(p->cd, &line, &line_len, &x, &len)) == (size_t)-1) 
 				return -po_failed_iconv;
 
 			x = p->buf;
diff --git a/src/poparser.h b/src/poparser.h
index 7d7b419..75f5367 100644
--- a/src/poparser.h
+++ b/src/poparser.h
@@ -70,7 +70,7 @@ struct po_parser {
 	bool first;
 	iconv_t cd;
 	enum po_entry previous;
-	int strcnt;
+	unsigned strcnt;
 	size_t max_ctxt_len;
 	size_t max_id_len;
 	size_t max_plural_len;
-- 
2.25.0

