From c6dcdcdef801127549d3906d153c061880d25a73 Mon Sep 17 00:00:00 2001
From: xhe <xw897002528@gmail.com>
Date: Tue, 5 Feb 2019 13:56:20 +0800
Subject: [PATCH 41/48] msgfmt: missing break

---
 src/msgfmt.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/msgfmt.c b/src/msgfmt.c
index 0352866..6f4cd40 100644
--- a/src/msgfmt.c
+++ b/src/msgfmt.c
@@ -414,8 +414,9 @@ int main(int argc, char**argv) {
 		}
 
 		ret = process(in, out, strict);
+		break;
 	default:
-		fprintf(stderr, "unsupported mode %d", mode);
+		fprintf(stderr, "unsupported mode %d\n", mode);
 		ret = -1;
 	}
 
-- 
2.25.0

