--- kernel/gen_kheaders.sh
+++ kernel/gen_kheaders.sh
@@ -87,7 +87,7 @@
 # pre-sorted, as --sort=name might not be available.
 find $cpio_dir -printf "./%P\n" | LC_ALL=C sort | \
     tar "${KBUILD_BUILD_TIMESTAMP:+--mtime=$KBUILD_BUILD_TIMESTAMP}" \
-    --owner=0 --group=0 --numeric-owner --no-recursion \
+    --gid 0 --uid 0 --numeric-owner --no-recursion \
     -Jcf $tarfile -C $cpio_dir/ -T - > /dev/null
 
 echo $headers_md5 > kernel/kheaders.md5
--- scripts/package/buildtar
+++ scripts/package/buildtar
@@ -133,8 +133,8 @@
 #
 # Create the tarball
 #
-if tar --owner=root --group=root --help >/dev/null 2>&1; then
-	opts="$opts --owner=root --group=root"
+if tar --gid 0 --uid 0 --help >/dev/null 2>&1; then
+	opts="$opts --gid 0 --uid 0"
 fi
 
 tar cf $tarball -C $tmpdir $opts $dirs
