--- configure
+++ configure
@@ -16004,8 +16004,8 @@
 
 
 case $host_cpu in
-	i?86*)   RESIZECONS_PROGS=yes ;;
-	x86_64*) RESIZECONS_PROGS=yes ;;
+	i?86*)   RESIZECONS_PROGS=no ;;
+	x86_64*) RESIZECONS_PROGS=no ;;
 	*)       RESIZECONS_PROGS=no  ;;
 esac
  if test "$RESIZECONS_PROGS" = "yes"; then
--- docs/man/man8/Makefile.in
+++ docs/man/man8/Makefile.in
@@ -314,7 +314,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 gen_MANS = loadunimap.8 mapscrn.8 setfont.8
-dist_man_MANS = getkeycodes.8 kbdrate.8 resizecons.8 setkeycodes.8 \
+dist_man_MANS = getkeycodes.8 kbdrate.8 setkeycodes.8 \
 		showconsolefont.8 setvtrgb.8 $(gen_MANS)
 
 all: all-am
