From fb7a162f535a12b8f79d6e2b7c3adbb720385297 Mon Sep 17 00:00:00 2001
From: Wayne Davison <wayned@samba.org>
Date: Sun, 25 Mar 2018 10:04:14 -0700
Subject: [PATCH 01/36] Prepare the repository for more development.

---
 NEWS         | 44 ++++----------------------------------------
 OLDNEWS      | 48 ++++++++++++++++++++++++++++++++++++++++++++++++
 configure.ac |  2 +-
 3 files changed, 53 insertions(+), 41 deletions(-)

diff --git a/NEWS b/NEWS
index e6239ff9..fd3c314d 100644
--- a/NEWS
+++ b/NEWS
@@ -1,47 +1,11 @@
-NEWS for rsync 3.1.3 (28 Jan 2018)
+NEWS for rsync 3.1.4 (UNRELEASED)
 Protocol: 31 (unchanged)
-Changes since 3.1.2:
-
-  SECURITY FIXES:
-    - Fixed a buffer overrun in the protocol's handling of xattr names and
-      ensure that the received name is null terminated.
-    - Fix an issue with --protect-args where the user could specify the arg in
-      the protected-arg list and short-circuit some of the arg-sanitizing code.
+Changes since 3.1.3:
 
   BUG FIXES:
 
-    - Don't output about a new backup dir without appropriate info verbosity.
-    - Fixed some issues with the sort functions in support/rsyncstats script.
-    - Added a way to specify daemon config lists (e.g. users, groups, etc) that
-      contain spaces (see "auth users" in the latest rsyncd.conf manpage).
-    - If a backup fails (e.g. full disk) rsync exits with an error.
-    - Fixed a problem with a doubled --fuzzy option combined with --link-dest.
-    - Avoid invalid output in the summary if either the start or end time had
-      an error.
-    - We don't allow a popt alias to affect the --daemon or --server options.
-    - Fix daemon exclude code to disallow attribute changes in addition to
-      disallowing transfers.
-    - Don't force nanoseconds to match if a non-transferred, non-checksummed
-      file only passed the quick-check w/o comparing nanosecods.
+    - ...
 
   ENHANCEMENTS:
 
-    - Added the ability for rsync to compare nanosecond times in its file-check
-      comparisons, and added support nanosecond times on Mac OS X.
-    - Added a short-option (-@) for --modify-window.
-    - Added the --checksum-choice=NAME[,NAME] option to choose the checksum
-      algorithms.
-    - Added hashing of xattr names (with using -X) to improve the handling of
-      files with large numbers of xattrs.
-    - Added a way to filter xattr names using include/exclude/filter rules (see
-      the --xattrs option in the manpage for details).
-    - Added "daemon chroot|uid|gid" to the daemon config (in addition to the
-      old chroot|uid|gid settings that affect the daemon's transfer process).
-    - Added "syslog tag" to the daemon configuration.
-    - Some manpage improvements.
-
-  DEVELOPER RELATED:
-
-    - Tweak the "make" output when yodl isn't around to create the man pages.
-    - Changed an obsolete autoconf compile macro.
-    - Support newer yodl versions when converting man pages.
+    - ...
diff --git a/OLDNEWS b/OLDNEWS
index 3665794d..7bc36eeb 100644
--- a/OLDNEWS
+++ b/OLDNEWS
@@ -1,3 +1,51 @@
+NEWS for rsync 3.1.3 (28 Jan 2018)
+Protocol: 31 (unchanged)
+Changes since 3.1.2:
+
+  SECURITY FIXES:
+    - Fixed a buffer overrun in the protocol's handling of xattr names and
+      ensure that the received name is null terminated.
+    - Fix an issue with --protect-args where the user could specify the arg in
+      the protected-arg list and short-circuit some of the arg-sanitizing code.
+
+  BUG FIXES:
+
+    - Don't output about a new backup dir without appropriate info verbosity.
+    - Fixed some issues with the sort functions in support/rsyncstats script.
+    - Added a way to specify daemon config lists (e.g. users, groups, etc) that
+      contain spaces (see "auth users" in the latest rsyncd.conf manpage).
+    - If a backup fails (e.g. full disk) rsync exits with an error.
+    - Fixed a problem with a doubled --fuzzy option combined with --link-dest.
+    - Avoid invalid output in the summary if either the start or end time had
+      an error.
+    - We don't allow a popt alias to affect the --daemon or --server options.
+    - Fix daemon exclude code to disallow attribute changes in addition to
+      disallowing transfers.
+    - Don't force nanoseconds to match if a non-transferred, non-checksummed
+      file only passed the quick-check w/o comparing nanosecods.
+
+  ENHANCEMENTS:
+
+    - Added the ability for rsync to compare nanosecond times in its file-check
+      comparisons, and added support nanosecond times on Mac OS X.
+    - Added a short-option (-@) for --modify-window.
+    - Added the --checksum-choice=NAME[,NAME] option to choose the checksum
+      algorithms.
+    - Added hashing of xattr names (with using -X) to improve the handling of
+      files with large numbers of xattrs.
+    - Added a way to filter xattr names using include/exclude/filter rules (see
+      the --xattrs option in the manpage for details).
+    - Added "daemon chroot|uid|gid" to the daemon config (in addition to the
+      old chroot|uid|gid settings that affect the daemon's transfer process).
+    - Added "syslog tag" to the daemon configuration.
+    - Some manpage improvements.
+
+  DEVELOPER RELATED:
+
+    - Tweak the "make" output when yodl isn't around to create the man pages.
+    - Changed an obsolete autoconf compile macro.
+    - Support newer yodl versions when converting man pages.
+
 NEWS for rsync 3.1.2 (21 Dec 2015)
 Protocol: 31 (unchanged)
 Changes since 3.1.1:
diff --git a/configure.ac b/configure.ac
index 07ce8002..45a96637 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1,6 +1,6 @@
 dnl Process this file with autoconf to produce a configure script.
 
-AC_INIT([rsync],[3.1.3],[http://rsync.samba.org/bugzilla.html])
+AC_INIT([rsync],[3.1.4dev],[http://rsync.samba.org/bugzilla.html])
 
 AC_CONFIG_MACRO_DIR([m4])
 AC_CONFIG_SRCDIR([byteorder.h])
-- 
2.25.0

