From 1a288c06d9c12c1d5fc754d6b95a2e61bf5a0250 Mon Sep 17 00:00:00 2001
From: Wayne Davison <wayned@samba.org>
Date: Tue, 20 Nov 2018 13:44:09 -0800
Subject: [PATCH 06/36] Document how a leading comma changes the gid parsing.

---
 rsyncd.conf.yo | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/rsyncd.conf.yo b/rsyncd.conf.yo
index 3076a492..c041cf93 100644
--- a/rsyncd.conf.yo
+++ b/rsyncd.conf.yo
@@ -408,6 +408,12 @@ supplementary groups.  The default for a non-super-user is to not change any
 group attributes (and indeed, your OS may not allow a non-super-user to try to
 change their group settings).
 
+The specified list is normally split into tokens based on spaces and commas.
+However, if the list starts with a comma, then the list is only split on
+commas, which allows a group name to contain a space.  In either case any
+leading and/or trailing whitespace is removed from the tokens and empty tokens
+are ignored.
+
 dit(bf(daemon uid)) This parameter specifies a uid under which the daemon will
 run. The daemon usually runs as user root, and when this is left unset the user
 is left unchanged. See also the "uid" parameter.
-- 
2.25.0

