From b4c1b27e03d470f757e6d3ff98172bad865ed6c3 Mon Sep 17 00:00:00 2001
From: Wayne Davison <wayned@samba.org>
Date: Tue, 8 Jan 2019 13:29:18 -0800
Subject: [PATCH 12/36] Fix 2 spelling errors pointed out by bug 13734.

---
 rsync.yo | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/rsync.yo b/rsync.yo
index 99177363..3a8f9cb5 100644
--- a/rsync.yo
+++ b/rsync.yo
@@ -1330,7 +1330,7 @@ batch-writing option is in effect.
 
 dit(bf(--checksum-choice=STR)) This option overrides the checksum algoriths.
 If one algorithm name is specified, it is used for both the transfer checksums
-and (assuming bf(--checksum) is specifed) the pre-transfer checksumming. If two
+and (assuming bf(--checksum) is specified) the pre-transfer checksumming. If two
 comma-separated names are supplied, the first name affects the transfer
 checksums, and the second name affects the pre-transfer checksumming.
 
@@ -2340,7 +2340,7 @@ units of 1024.
 
 The default is human-readable level 1.  Each bf(-h) option increases the level
 by one.  You can take the level down to 0 (to output numbers as pure digits) by
-specifing the bf(--no-human-readable) (bf(--no-h)) option.
+specifying the bf(--no-human-readable) (bf(--no-h)) option.
 
 The unit letters that are appended in levels 2 and 3 are: K (kilo), M (mega),
 G (giga), or T (tera).  For example, a 1234567-byte file would output as 1.23M
-- 
2.25.0

