From 48163179ebb1cc79a207a8cd80bb11c1f455bc00 Mon Sep 17 00:00:00 2001
From: Wayne Davison <wayned@samba.org>
Date: Tue, 8 Jan 2019 13:38:19 -0800
Subject: [PATCH 13/36] Avoid a yodl macro warning.

---
 rsync.yo | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rsync.yo b/rsync.yo
index 3a8f9cb5..7bf005cc 100644
--- a/rsync.yo
+++ b/rsync.yo
@@ -238,7 +238,7 @@ which forwards all data to port 873 (the rsync daemon) on the targethost
 
 Note also that if the RSYNC_SHELL environment varibable is set, that
 program will be used to run the RSYNC_CONNECT_PROG command instead of
-using the default shell of the system() call.
+using the default shell of the code(system()) call.
 
 manpagesection(USING RSYNC-DAEMON FEATURES VIA A REMOTE-SHELL CONNECTION)
 
-- 
2.25.0

