From e5610f1877ebb2a3fa7fb864d78cf41d0246701d Mon Sep 17 00:00:00 2001
From: Wayne Davison <wayned@samba.org>
Date: Tue, 8 Jan 2019 16:39:48 -0800
Subject: [PATCH 15/36] Avoid a yodl macro warning.

---
 rsyncd.conf.yo | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/rsyncd.conf.yo b/rsyncd.conf.yo
index c041cf93..8f004ae6 100644
--- a/rsyncd.conf.yo
+++ b/rsyncd.conf.yo
@@ -83,7 +83,7 @@ default for that parameter.
 
 You may use references to environment variables in the values of parameters.
 String parameters will have %VAR% references expanded as late as possible (when
-the string is used in the program), allowing for the use of variables that
+the string is first used in the program), allowing for the use of variables that
 rsync sets at connection time, such as RSYNC_USER_NAME.  Non-string parameters
 (such as true/false settings) are expanded when read from the config file.  If
 a variable does not exist in the environment, or if a sequence of characters is
@@ -819,7 +819,7 @@ are run using the permissions of the user that started the daemon (not the
 module's uid/gid setting) without any chroot restrictions.
 
 These settings honor 2 environment variables: use RSYNC_SHELL to set a shell to
-use when running the command (which otherwise uses your system() call's default
+use when running the command (which otherwise uses your code(system()) call's default
 shell), and use RSYNC_NO_XFER_EXEC to disable both options completely.
 
 )
-- 
2.25.0

