From 48346c878f03868480c6c8db1f41e69576758ccc Mon Sep 17 00:00:00 2001
From: Wayne Davison <wayned@samba.org>
Date: Wed, 9 Jan 2019 12:15:31 -0800
Subject: [PATCH 19/36] Reject --log-file when read-only.

---
 support/rrsync | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/support/rrsync b/support/rrsync
index 6079c95d..ad4d434b 100644
--- a/support/rrsync
+++ b/support/rrsync
@@ -107,7 +107,7 @@ our %long_opt = (
   'link-dest' => 2,
   'links' => 0,
   'list-only' => 0,
-  'log-file' => 3,
+  'log-file' => $only eq 'r' ? -1 : 3,
   'log-format' => 1,
   'max-delete' => 1,
   'max-size' => 1,
-- 
2.25.0

