From bc7402aa3a1148c02025b734a394198cf595a3a5 Mon Sep 17 00:00:00 2001
From: Wayne Davison <wayned@samba.org>
Date: Tue, 15 Jan 2019 10:40:18 -0800
Subject: [PATCH 22/36] Avoid warning about leaked mem (didn't affect rsync's
 pool use).

---
 lib/pool_alloc.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/lib/pool_alloc.c b/lib/pool_alloc.c
index 5856d591..a70a3f1a 100644
--- a/lib/pool_alloc.c
+++ b/lib/pool_alloc.c
@@ -49,15 +49,15 @@ pool_create(size_t size, size_t quantum, void (*bomb)(const char *), int flags)
 {
 	struct alloc_pool *pool;
 
-	if (!(pool = new0(struct alloc_pool)))
-		return NULL;
-
 	if ((MINALIGN & (MINALIGN - 1)) != 0) {
 		if (bomb)
 			(*bomb)("Compiler error: MINALIGN is not a power of 2\n");
 		return NULL;
 	}
 
+	if (!(pool = new0(struct alloc_pool)))
+		return NULL;
+
 	if (!size)
 		size = POOL_DEF_EXTENT;
 	if (!quantum)
-- 
2.25.0

