From 8149ea52bd4968dbaa99083c2827a0cbb12cfc17 Mon Sep 17 00:00:00 2001
From: William Douglas <william.douglas@intel.com>
Date: Tue, 17 Nov 2015 23:03:30 +0000
Subject: [PATCH] Lower test timeout

---
 Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index 320e86f..8f43d48 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -393,7 +393,7 @@ EXTRA_DIST += grub-core/tests/boot/kbsd.init-i386.S grub-core/tests/boot/kbsd.in
 # Randomly generated
 SUCCESSFUL_BOOT_STRING=3e49994fd5d82b7c9298d672d774080d
 # tianocore cd access is very slow
-BOOTCHECK_TIMEOUT=180
+BOOTCHECK_TIMEOUT=3
 
 bootcheck: $(BOOTCHECKS)
 
-- 
2.6.3

