From 73ee3a05785ac53afbf6128c8b13010f66491320 Mon Sep 17 00:00:00 2001
From: Auke Kok <auke-jan.h.kok@intel.com>
Date: Mon, 2 Apr 2018 13:38:53 -0700
Subject: [PATCH 1/4] Speed up agetty waits.

---
 term-utils/agetty.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/term-utils/agetty.c b/term-utils/agetty.c
index 8442204..99bb59d 100644
--- a/term-utils/agetty.c
+++ b/term-utils/agetty.c
@@ -24,6 +24,7 @@
 #include <sys/wait.h>
 #include <fcntl.h>
 #include <stdarg.h>
+#include <malloc.h>
 #include <ctype.h>
 #include <utmpx.h>
 #include <getopt.h>
@@ -1390,7 +1391,7 @@ static void auto_baud(struct termios *tp)
 	 * Wait for a while, then read everything the modem has said so far and
 	 * try to extract the speed of the dial-in call.
 	 */
-	sleep(1);
+	usleep(1000);
 	if ((nread = read(STDIN_FILENO, buf, sizeof(buf) - 1)) > 0) {
 		buf[nread] = '\0';
 		for (bp = buf; bp < buf + nread; bp++)
@@ -1907,6 +1908,7 @@ again:
 		/* Always show login prompt. */
 		write_all(STDOUT_FILENO, LOGIN, sizeof(LOGIN) - 1);
 	}
+	malloc_trim(0);
 }
 
 /* Select next baud rate. */
@@ -1952,7 +1954,7 @@ static char *get_logname(struct options *op, struct termios *tp, struct chardata
 	 * the baud rate).
 	 */
 	if ((op->flags & F_VCONSOLE) == 0)
-		sleep(1);
+		usleep(1000);
 	tcflush(STDIN_FILENO, TCIFLUSH);
 
 	eightbit = (op->flags & F_EIGHTBITS);
-- 
2.16.3

