--- configure
+++ configure
@@ -660,11 +660,11 @@
 test "$ARCH" = "microblaze" && trycppif __MICROBLAZEEL__ "$t" \
 && SUBARCH=${SUBARCH}el
 
-if test "$ARCH" = "powerpc64" ; then
-trycppif "_CALL_ELF == 2" "$t" || fail "$0: error: unsupported powerpc64 ABI"
-trycppif __LITTLE_ENDIAN__ "$t" && SUBARCH=${SUBARCH}le
-trycppif _SOFT_FLOAT "$t" && fail "$0: error: soft-float not supported on powerpc64"
-fi
+#if test "$ARCH" = "powerpc64" ; then
+#trycppif "_CALL_ELF == 2" "$t" || fail "$0: error: unsupported powerpc64 ABI"
+#trycppif __LITTLE_ENDIAN__ "$t" && SUBARCH=${SUBARCH}le
+#trycppif _SOFT_FLOAT "$t" && fail "$0: error: soft-float not supported on powerpc64"
+#fi
 
 if test "$ARCH" = "riscv64" ; then
 trycppif __riscv_float_abi_soft "$t" && SUBARCH=${SUBARCH}-sf
