--- GNUmakefile
+++ GNUmakefile
@@ -378,7 +378,7 @@
 tabs/%.o: tabs/%.c
 	$(CC) $(CPPFLAGS) -I. -I./libterminfo $(CFLAGS) -c -o $@ $<
 nbperf/%.o: nbperf/%.c
-	$(HOSTCC) $(CPPFLAGS) -O0 -g0 $(CFLAGS_HOST) -c -o $@ $<
+	$(HOSTCC) $(CPPFLAGS) -g0 $(CFLAGS_HOST) -c -o $@ $<
 
 ifeq ($(CROSSCOMPILING),0)
 tic/%.ho: tic/%.o
@@ -388,7 +388,7 @@
 	$(LN) -sf $(notdir $<) $@
 else
 tic/%.ho: tic/%.c
-	$(HOSTCC) $(CPPFLAGS) -I./tic -O0 -g0 $(CFLAGS_HOST) -c -o $@ $<
+	$(HOSTCC) $(CPPFLAGS) -I./tic -g0 $(CFLAGS_HOST) -c -o $@ $<
 
 $(TOOL_HOSTTIC): $(TIC_HOBJS)
 	$(HOSTCC) $(LDFLAGS_HOST) $^ -o $@
