diff -urNa e2fsprogs-1.44.4.orig/e2fsck/problem.c e2fsprogs-1.44.4/e2fsck/problem.c
--- e2fsprogs-1.44.4.orig/e2fsck/problem.c	2018-08-19 05:26:58.000000000 +0300
+++ e2fsprogs-1.44.4/e2fsck/problem.c	2018-08-29 16:28:15.063458587 +0300
@@ -99,7 +99,7 @@
 	"",			/* 20 */
 };
 
-#if __GNUC_PREREQ (4, 6)
+#if defined(__GNUC__) && ((__GNUC__<<16)+__GNUC_MINOR__) >= 0x406
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wmissing-field-initializers"
 #endif
@@ -2094,7 +2094,7 @@
 	{ PR_LATCH_OPTIMIZE_EXT, PR_1E_OPTIMIZE_EXT_HEADER, PR_1E_OPTIMIZE_EXT_END },
 	{ -1, 0, 0 },
 };
-#if __GNUC_PREREQ (4, 6)
+#if defined(__GNUC__) && ((__GNUC__<<16)+__GNUC_MINOR__) >= 0x406
 #pragma GCC diagnostic pop
 #endif
 
diff -urNa e2fsprogs-1.44.4.orig/lib/ext2fs/ext4_acl.h e2fsprogs-1.44.4/lib/ext2fs/ext4_acl.h
--- e2fsprogs-1.44.4.orig/lib/ext2fs/ext4_acl.h	2018-08-19 05:26:58.000000000 +0300
+++ e2fsprogs-1.44.4/lib/ext2fs/ext4_acl.h	2018-08-29 16:25:50.103449775 +0300
@@ -50,12 +50,12 @@
 
 typedef struct {
         __le32                  a_version;
-#if __GNUC_PREREQ (4, 8)
+#if defined(__GNUC__) && ((__GNUC__<<16)+__GNUC_MINOR__) >= 0x408
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wpedantic"
 #endif
         posix_acl_xattr_entry   a_entries[0];
-#if __GNUC_PREREQ (4, 8)
+#if defined(__GNUC__) && ((__GNUC__<<16)+__GNUC_MINOR__) >= 0x408
 #pragma GCC diagnostic pop
 #endif
 } posix_acl_xattr_header;

diff -urNa e2fsprogs-1.44.4.orig/lib/ext2fs/fiemap.h e2fsprogs-1.44.4/lib/ext2fs/fiemap.h
--- e2fsprogs-1.44.4.orig/lib/ext2fs/fiemap.h	2018-08-19 05:26:58.000000000 +0300
+++ e2fsprogs-1.44.4/lib/ext2fs/fiemap.h	2018-08-29 16:29:04.796794928 +0300
@@ -31,12 +31,12 @@
 	__u32 fm_mapped_extents;/* number of extents that were mapped (out) */
 	__u32 fm_extent_count;  /* size of fm_extents array (in) */
 	__u32 fm_reserved;
-#if __GNUC_PREREQ (4, 8)
+#if defined(__GNUC__) && ((__GNUC__<<16)+__GNUC_MINOR__) >= 0x408
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wpedantic"
 #endif
 	struct fiemap_extent fm_extents[0]; /* array of mapped extents (out) */
-#if __GNUC_PREREQ (4, 8)
+#if defined(__GNUC__) && ((__GNUC__<<16)+__GNUC_MINOR__) >= 0x408
 #pragma GCC diagnostic pop
 #endif
 };

diff -urNa e2fsprogs-1.44.4.orig/lib/ext2fs/hashmap.h e2fsprogs-1.44.4/lib/ext2fs/hashmap.h
--- e2fsprogs-1.44.4.orig/lib/ext2fs/hashmap.h	2018-08-19 05:26:58.000000000 +0300
+++ e2fsprogs-1.44.4/lib/ext2fs/hashmap.h	2018-08-29 16:23:31.236774667 +0300
@@ -17,12 +17,12 @@
 		struct ext2fs_hashmap_entry *next;
 		struct ext2fs_hashmap_entry *list_next;
 		struct ext2fs_hashmap_entry *list_prev;
-#if __GNUC_PREREQ (4, 8)
+#if defined(__GNUC__) &&  ((__GNUC__<<16)+__GNUC_MINOR__) >= 0x408
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wpedantic"
 #endif
 	} *entries[0];
-#if __GNUC_PREREQ (4, 8)
+#if defined(__GNUC__) &&  ((__GNUC__<<16)+__GNUC_MINOR__) >= 0x408
 #pragma GCC diagnostic pop
 #endif
 };

diff -urNa e2fsprogs-1.44.4.orig/lib/ext2fs/mmp.c e2fsprogs-1.44.4/lib/ext2fs/mmp.c
--- e2fsprogs-1.44.4.orig/lib/ext2fs/mmp.c	2018-08-19 05:26:58.000000000 +0300
+++ e2fsprogs-1.44.4/lib/ext2fs/mmp.c	2018-08-29 16:26:42.163452935 +0300
@@ -34,7 +34,7 @@
 #define O_DIRECT 0
 #endif
 
-#if __GNUC_PREREQ (4, 6)
+#if defined(__GNUC__) && ((__GNUC__<<16)+__GNUC_MINOR__) >= 0x406
 #pragma GCC diagnostic push
 #ifndef CONFIG_MMP
 #pragma GCC diagnostic ignored "-Wunused-parameter"
@@ -467,6 +467,6 @@
 	return EXT2_ET_OP_NOT_SUPPORTED;
 #endif
 }
-#if __GNUC_PREREQ (4, 6)
+#if defined(__GNUC__) && ((__GNUC__<<16)+__GNUC_MINOR__) >= 0x406
 #pragma GCC diagnostic pop
 #endif

diff -urNa e2fsprogs-1.44.4.orig/lib/ext2fs/unix_io.c e2fsprogs-1.44.4/lib/ext2fs/unix_io.c
--- e2fsprogs-1.44.4.orig/lib/ext2fs/unix_io.c	2018-08-19 05:26:58.000000000 +0300
+++ e2fsprogs-1.44.4/lib/ext2fs/unix_io.c	2018-08-29 16:27:27.586789035 +0300
@@ -1127,7 +1127,7 @@
 }
 
 /* parameters might not be used if OS doesn't support zeroout */
-#if __GNUC_PREREQ (4, 6)
+#if defined(__GNUC__) && ((__GNUC__<<16)+__GNUC_MINOR__) >= 0x406
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wunused-parameter"
 #endif
@@ -1197,7 +1197,7 @@
 unimplemented:
 	return EXT2_ET_UNIMPLEMENTED;
 }
-#if __GNUC_PREREQ (4, 6)
+#if defined(__GNUC__) && ((__GNUC__<<16)+__GNUC_MINOR__) >= 0x406
 #pragma GCC diagnostic pop
 #endif
 
diff -urNa e2fsprogs-1.44.4.orig/lib/uuid/gen_uuid.c e2fsprogs-1.44.4/lib/uuid/gen_uuid.c
--- e2fsprogs-1.44.4.orig/lib/uuid/gen_uuid.c	2018-08-19 05:26:58.000000000 +0300
+++ e2fsprogs-1.44.4/lib/uuid/gen_uuid.c	2018-08-29 16:24:51.203446187 +0300
@@ -484,7 +484,7 @@
 }
 #endif /* defined(USE_UUIDD) && defined(HAVE_SYS_UN_H) */
 
-#if __GNUC_PREREQ (4, 6)
+#if defined(__GNUC__) && ((__GNUC__<<16)+__GNUC_MINOR__) >= 0x406
 #pragma GCC diagnostic push
 #if !defined(USE_UUIDD) || !defined(HAVE_SYS_UN_H)
 #pragma GCC diagnostic ignored "-Wunused-parameter"
@@ -572,7 +572,7 @@
 #endif
 	return -1;
 }
-#if __GNUC_PREREQ (4, 6)
+#if defined(__GNUC__) && ((__GNUC__<<16)+__GNUC_MINOR__) >= 0x406
 #pragma GCC diagnostic pop
 #endif
 
diff -urNa e2fsprogs-1.44.4.orig/misc/e2undo.c e2fsprogs-1.44.4/misc/e2undo.c
--- e2fsprogs-1.44.4.orig/misc/e2undo.c	2018-08-19 05:26:58.000000000 +0300
+++ e2fsprogs-1.44.4/misc/e2undo.c	2018-08-29 16:31:10.486802549 +0300
@@ -81,12 +81,12 @@
 	__le32 magic;		/* KEYBLOCK_MAGIC number */
 	__le32 crc;		/* block checksum */
 	__le64 reserved;	/* zero */
-#if __GNUC_PREREQ (4, 8)
+#if defined(__GNUC__) && ((__GNUC__<<16)+__GNUC_MINOR__) >= 0x408
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wpedantic"
 #endif
 	struct undo_key keys[0];	/* keys, which come immediately after */
-#if __GNUC_PREREQ (4, 8)
+#if defined(__GNUC__) && ((__GNUC__<<16)+__GNUC_MINOR__) >= 0x408
 #pragma GCC diagnostic pop
 #endif
 };

diff -urNa e2fsprogs-1.44.4.orig/misc/filefrag.c e2fsprogs-1.44.4/misc/filefrag.c
--- e2fsprogs-1.44.4.orig/misc/filefrag.c	2018-08-19 05:26:58.000000000 +0300
+++ e2fsprogs-1.44.4/misc/filefrag.c	2018-08-29 16:31:44.673471284 +0300
@@ -535,7 +535,7 @@
 				char *end;
 				blocksize = strtoul(optarg, &end, 0);
 				if (end) {
-#if __GNUC_PREREQ (7, 0)
+#if defined(__GNUC__) && ((__GNUC__<<16)+__GNUC_MINOR__) >= 0x700
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wimplicit-fallthrough"
 #endif
@@ -555,7 +555,7 @@
 					default:
 						break;
 					}
-#if __GNUC_PREREQ (7, 0)
+#if defined(__GNUC__) && ((__GNUC__<<16)+__GNUC_MINOR__) >= 0x700
 #pragma GCC diagnostic pop
 #endif
 				}
