From 5fbd3f8213121d4cf550e712ee2bd3a5d7b4a276 Mon Sep 17 00:00:00 2001
From: Paul Zimmermann <Paul.Zimmermann@inria.fr>
Date: Fri, 8 Jun 2018 18:03:14 +0200
Subject: [PATCH 13/29] [tools/coverage] added branch coverage

---
 tools/coverage | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/tools/coverage b/tools/coverage
index 5bc4c3c..6a1d8b8 100755
--- a/tools/coverage
+++ b/tools/coverage
@@ -40,7 +40,7 @@ autoreconf -fi || exit 1
 
 echo "Building MPC"
 ./configure --disable-shared --enable-static \
-  CFLAGS="-fprofile-arcs -ftest-coverage -g"  || exit 1
+  CFLAGS="--coverage -g"  || exit 1
 make clean  || exit 1
 make all -j4 || exit 1
 
@@ -115,8 +115,8 @@ done | bash ../coverage.subscript | grep -v '__gmp' > ../coverage.mpc
 rm -f coverage.subscript coverage-tmp || exit 1
 
 cd -
-lcov --capture --directory . --output-file all.info || exit 1
-genhtml -o coverage all.info || exit 1
+lcov --rc lcov_branch_coverage=1 --capture --directory . --output-file all.info || exit 1
+genhtml --branch-coverage -o coverage all.info || exit 1
 
 echo "Coverage summary saved in file    /tmp/ompc-gcov/coverage.mpc"
 echo "Detailed coverage is available at /tmp/ompc-gcov/coverage/index.html"
-- 
2.25.0

