From 7d0dbdba2526d6b60bc45ceb33bc8bb44dcb64b5 Mon Sep 17 00:00:00 2001
From: Paul Zimmermann <Paul.Zimmermann@inria.fr>
Date: Thu, 21 Nov 2019 19:07:09 +0100
Subject: [PATCH 28/29] updated the build instructions

---
 README.dev | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/README.dev b/README.dev
index 3f0b9c6..48c8d42 100644
--- a/README.dev
+++ b/README.dev
@@ -1,10 +1,14 @@
-Source code from the subversion repository
-------------------------------------------
+Source code from the git repository
+-----------------------------------
 
-To compile source code obtained from the subversion repository,
+To compile source code obtained from the git repository,
 make sure you have installed the autoconf, automake and libtool packages.
-Execute
+Execute:
+
 $ autoreconf -i
+$ ./configure
+$ make
+$ make check
 
 Required versions:
 * autoconf >= 2.61
-- 
2.25.0

