From 99421f82e544e7192d629767eb066505df147870 Mon Sep 17 00:00:00 2001
From: Paul Zimmermann <Paul.Zimmermann@inria.fr>
Date: Thu, 23 Jan 2020 13:50:54 +0100
Subject: [PATCH 29/29] added reference to Pascal Molin paper on erf

---
 TODO | 1 +
 1 file changed, 1 insertion(+)

diff --git a/TODO b/TODO
index 662ecc2..6ce93a6 100644
--- a/TODO
+++ b/TODO
@@ -71,6 +71,7 @@ New functions to implement:
 - from Joseph S. Myers <joseph at codesourcery dot com> 19 Mar 2012: mpc_erf,
   mpc_erfc, mpc_exp2, mpc_expm1, mpc_log1p, mpc_log2, mpc_lgamma, mpc_tgamma
   http://lists.gforge.inria.fr/pipermail/mpc-discuss/2012-March/001090.html
+  See the article by Pascal Molin (hal.archives-ouvertes.fr/hal-00580855).
 - from Andreas Enge and Philippe Théveny 17 July 2008
   agm (and complex logarithm with agm ?). For the error analysis, one can
   start from Theorem 1 of http://www.lix.polytechnique.fr/Labo/Regis.Dupont/preprints/Dupont_FastEvalMod.ps.gz, and probably the best is to compute AGM(a,b)
-- 
2.25.0

