diff -urNa lilo-24.2.orig/Makefile lilo-24.2/Makefile
--- lilo-24.2.orig/Makefile	2015-11-22 01:46:57.000000000 +0200
+++ lilo-24.2/Makefile	2018-08-25 22:42:13.011131319 +0300
@@ -12,7 +12,6 @@
 #
 all: test
 	make -C src all
-	make -C images all
 
 #
 #  make help
@@ -45,7 +44,6 @@
 #
 alles: test
 	$(MAKE) -C src alles
-	$(MAKE) -C images all
 
 #
 # documentation files
@@ -104,12 +102,8 @@
 #
 install:  all
 	$(MAKE) -C src install
-	$(MAKE) -C images install
-	$(MAKE) -C hooks install
-	$(MAKE) -C sample install
 	$(MAKE) -C scripts install
 	$(MAKE) -C man install
-	$(MAKE) -C dos install
 
 tidy:
 	$(MAKE) -C src tidy
diff -urNa lilo-24.2.orig/make.vars lilo-24.2/make.vars
--- lilo-24.2.orig/make.vars	2013-06-07 10:50:30.000000000 +0300
+++ lilo-24.2/make.vars	2018-08-25 22:46:52.157815123 +0300
@@ -70,8 +70,8 @@
 #   XL_SECS=n   Support for extra large (non-standard) floppies.
 
 CONFIG=-DBDATA -DDSECS=3 -DEVMS -DIGNORECASE -DLVM -DNOKEYBOARD -DONE_SHOT \
-    -DPASS160 -DREISERFS -DREWRITE_TABLE -DSOLO_CHAIN -DVERSION -DVIRTUAL \
-    -DMDPRAID -DDEVMAPPER
+    -DPASS160 -DREISERFS -DREWRITE_TABLE -DSOLO_CHAIN -DVIRTUAL \
+    -DMDPRAID
    
 # set the following if you wish LILO.COM to be installed
 DOS_DIR=/dosC/boot
@@ -100,8 +100,6 @@
   sed 's/-D/-DLCF_/g'`
 LILO=-DLILO=0xbb920890
 KVER=`if [ -f /usr/include/linux/version.h ]; then echo -DHAS_VERSION_H; fi`
-HDM=`if [ -f /usr/include/libdevmapper.h ]; then echo -DHAS_LIBDEVMAPPER_H; fi`
-DEVMAPPER=`if [ -f /usr/include/libdevmapper.h ]; then echo "-ldevmapper"; fi`
 
 TMP=_TmP_
 
diff -urNa lilo-24.2.orig/man/Makefile lilo-24.2/man/Makefile
--- lilo-24.2.orig/man/Makefile	2014-10-15 11:14:57.000000000 +0300
+++ lilo-24.2/man/Makefile	2018-08-25 22:43:41.304470073 +0300
@@ -12,7 +12,7 @@
 
 PODFILES  := lilo-uuid-diskid lilo liloconfig
 MANFILES5 := lilo.conf
-MANFILES8 := keytab-lilo lilo-uuid-diskid lilo liloconfig mkrescue
+MANFILES8 := keytab-lilo lilo-uuid-diskid lilo
 
 all:
 
diff -urNa lilo-24.2.orig/scripts/Makefile lilo-24.2/scripts/Makefile
--- lilo-24.2.orig/scripts/Makefile	2014-10-17 00:53:29.000000000 +0300
+++ lilo-24.2/scripts/Makefile	2018-08-25 22:44:05.981138255 +0300
@@ -15,7 +15,6 @@
 install:
 	mkdir -p $$DESTDIR$(USRSBIN_DIR)
 
-	install -m 0755  liloconfig  $$DESTDIR$(USRSBIN_DIR)
 	install -m 0755  lilo-uuid-diskid  $$DESTDIR$(USRSBIN_DIR)
 
 clean:
diff -urNa lilo-24.2.orig/src/Makefile lilo-24.2/src/Makefile
--- lilo-24.2.orig/src/Makefile	2015-11-22 01:50:25.000000000 +0200
+++ lilo-24.2/src/Makefile	2018-08-25 22:45:28.214476636 +0300
@@ -47,7 +47,7 @@
 #
 # everything needed to run, just short of installation
 #
-all:	lilo bootsect.b
+all:	lilo
 
 #
 # everything above plus the statically linked version
@@ -248,7 +248,6 @@
 	  strip lilo.static; \
 	  cp lilo.static $$DESTDIR$(SBIN_DIR)/lilo.static; \
 	fi
-	cp ../mkrescue $$DESTDIR$(USRSBIN_DIR)/mkrescue
 	cp ../keytab-lilo.pl $$DESTDIR$(USRSBIN_DIR)/keytab-lilo
 
 dep:
diff -urNa lilo-24.2.orig/src/lilo.c lilo-24.2/src/lilo.c
--- lilo-24.2.orig/src/lilo.c	2015-11-22 01:51:24.000000000 +0200
+++ lilo-24.2/src/lilo.c	2018-08-25 22:47:41.951151512 +0300
@@ -182,7 +182,6 @@
                 printf("Without");
 #endif
                 printf(" device-mapper\n");
-		printf("\nglibc version %d.%d\n", __GLIBC__, __GLIBC_MINOR__);
 		printf("Kernel Headers included from  %d.%d.%d\n",
 			LINUX_VERSION_CODE>>16,
 			LINUX_VERSION_CODE>>8 & 255,
