From b3d3ea6456a7c1ed174c332eb718ca042dbf6be8 Mon Sep 17 00:00:00 2001
From: Sebastian Pipping <sebastian@pipping.org>
Date: Sat, 28 Sep 2019 15:16:09 +0200
Subject: [PATCH 01/41] Changes: Fix spelling of "September"

---
 expat/Changes | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/expat/Changes b/expat/Changes
index 3409471..3bee5fd 100644
--- a/expat/Changes
+++ b/expat/Changes
@@ -2,7 +2,7 @@ NOTE: We are looking for help with a few things:
       https://github.com/libexpat/libexpat/labels/help%20wanted
       If you can help, please get in touch.  Thanks!
 
-Release 2.2.9 Wed Septemper 25 2019
+Release 2.2.9 Wed September 25 2019
         Other changes:
                   examples: Drop executable bits from elements.c
             #349  Windows: Change the name of the Windows DLLs from expat*.dll
@@ -17,7 +17,7 @@ Release 2.2.9 Wed Septemper 25 2019
         Special thanks to:
             Ben Wagner
 
-Release 2.2.8 Fri Septemper 13 2019
+Release 2.2.8 Fri September 13 2019
         Security fixes:
        #317 #318  CVE-2019-15903 -- Fix heap overflow triggered by
                     XML_GetCurrentLineNumber (or XML_GetCurrentColumnNumber),
-- 
2.25.0

