From c675d2d9da50d5727741259f6be522e31d7f8c1b Mon Sep 17 00:00:00 2001
From: Sebastian Pipping <sebastian@pipping.org>
Date: Tue, 1 Oct 2019 21:46:14 +0200
Subject: [PATCH 09/41] Changes: Document #354

---
 expat/Changes | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/expat/Changes b/expat/Changes
index 3bee5fd..1bb41a0 100644
--- a/expat/Changes
+++ b/expat/Changes
@@ -2,6 +2,11 @@ NOTE: We are looking for help with a few things:
       https://github.com/libexpat/libexpat/labels/help%20wanted
       If you can help, please get in touch.  Thanks!
 
+Release x.x.x xxx xxxxxxxxx xx xxxx
+        Other changes:
+            #354  Fix testsuite for -DEXPAT_DTD=OFF and -DEXPAT_NS=OFF, each
+            #354  Address compiler warnings
+
 Release 2.2.9 Wed September 25 2019
         Other changes:
                   examples: Drop executable bits from elements.c
-- 
2.25.0

