From a0ce01d3b0d5827538a856656918a2ac8cb74dfc Mon Sep 17 00:00:00 2001
From: Sebastian Pipping <sebastian@pipping.org>
Date: Sat, 12 Oct 2019 23:20:52 +0200
Subject: [PATCH 22/41] Changes: Document #356

---
 expat/Changes | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/expat/Changes b/expat/Changes
index e7b1703..a7ef645 100644
--- a/expat/Changes
+++ b/expat/Changes
@@ -5,10 +5,11 @@ NOTE: We are looking for help with a few things:
 Release x.x.x xxx xxxxxxxxx xx xxxx
         Other changes:
             #354  Fix testsuite for -DEXPAT_DTD=OFF and -DEXPAT_NS=OFF, each
-       #354 #355  Address compiler warnings
+  #354 #355 #356  Address compiler warnings
 
         Special thanks to:
             Ma Lin
+            Vadim Zeitlin
 
 Release 2.2.9 Wed September 25 2019
         Other changes:
-- 
2.25.0

