From b65e0d383d1430d3d8bf68554085a2f8a8e70b71 Mon Sep 17 00:00:00 2001
From: Sebastian Pipping <sebastian@pipping.org>
Date: Sat, 12 Oct 2019 23:22:17 +0200
Subject: [PATCH 28/41] Changes: Document improvements to CMake

---
 expat/Changes | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/expat/Changes b/expat/Changes
index a7ef645..415e9f2 100644
--- a/expat/Changes
+++ b/expat/Changes
@@ -4,6 +4,12 @@ NOTE: We are looking for help with a few things:
 
 Release x.x.x xxx xxxxxxxxx xx xxxx
         Other changes:
+            #360  CMake: Create libexpatw.{dll,so} and expatw.pc (with emphasis
+                    on suffix "w") with -DEXPAT_CHAR_TYPE=(ushort|wchar_t)
+            #360  CMake: Detect and deny unsupported build combinations
+                    involving -DEXPAT_CHAR_TYPE=(ushort|wchar_t)
+            #360  CMake: Install pre-compiled shipped xmlwf.1 manpage in case
+                    of -DEXPAT_BUILD_DOCS=OFF
             #354  Fix testsuite for -DEXPAT_DTD=OFF and -DEXPAT_NS=OFF, each
   #354 #355 #356  Address compiler warnings
 
-- 
2.25.0

