From 739a525d1b27008435f029d3188fb46f45af3f2c Mon Sep 17 00:00:00 2001
From: Sebastian Pipping <sebastian@pipping.org>
Date: Wed, 16 Oct 2019 15:23:14 +0200
Subject: [PATCH 31/41] Changes: Give credit for #361 and #362

---
 expat/Changes | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/expat/Changes b/expat/Changes
index 415e9f2..51eefe4 100644
--- a/expat/Changes
+++ b/expat/Changes
@@ -14,7 +14,9 @@ Release x.x.x xxx xxxxxxxxx xx xxxx
   #354 #355 #356  Address compiler warnings
 
         Special thanks to:
+            asavah
             Ma Lin
+            Mohammed Khajapasha
             Vadim Zeitlin
 
 Release 2.2.9 Wed September 25 2019
-- 
2.25.0

