From a6906cfd818267ff63d7aa37209c38a33f4643b2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C3=98rjan=20Malde?= <red@foxi.me>
Date: Tue, 31 Dec 2019 16:42:30 +0100
Subject: [PATCH 4/9] portable.h: midipix port

---
 portable.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/portable.h b/portable.h
index 9af6f79..84d93e3 100644
--- a/portable.h
+++ b/portable.h
@@ -9,14 +9,14 @@
  * Includes
  */
 
-#if defined(__linux__) || defined(__CYGWIN__)
+#if defined(__linux__) || defined(__CYGWIN__) || defined(__midipix__)
 #include <sys/file.h>
 #include <sys/types.h>
 
 #include <grp.h>
 #include <stdint.h>
 #include <stdlib.h>
-#endif /* __linux__ */
+#endif /* __linux__ || __CYGWIN__ || __midipix__ */
 
 #include <sys/param.h>
 #include <sys/time.h>
@@ -47,7 +47,7 @@
 #endif /* !O_EXLOCK */
 
 #ifndef _PW_NAME_LEN
-#if defined(__linux__) || defined(__CYGWIN__) || defined(_AIX)
+#if defined(__linux__) || defined(__CYGWIN__) || defined(_AIX) || defined(__midipix__)
 #define _PW_NAME_LEN	LOGIN_NAME_MAX
 #elif defined(__NetBSD__)
 #define _PW_NAME_LEN	MAXLOGNAME
@@ -55,7 +55,7 @@
 #define _PW_NAME_LEN	LOGNAME_MAX
 #else
 #define _PW_NAME_LEN	MAXLOGNAME - 1
-#endif /* __linux__ || __CYGWIN__ || _AIX || __NetBSD__ || __sun */
+#endif /* __linux__ || __CYGWIN__ || _AIX || __NetBSD__ || __sun || __midipix__ */
 #endif /* !_PW_NAME_LEN */
 
 #ifndef RLIMIT_RSS
-- 
2.25.0

